services/app-api/types.ts

```diff
@@ -32,6 +43,22 @@ export interface Measure {
    * it is set on fetch, according to the metadata in measureList.ts.
    */
   autoCompleted?: boolean;
+  data?: {
+    DataSource?: DataSource[];
+    DataSourceSelections?: unknown;
+    PerformanceMeasure?: {
+      rates?: {
+        [key: string]: StandardRateShape[];
```

Comment:
This type is not fully correct; in addition to being an array of StandardRateShapes, these values can be an array of what we might call FieldRateShapes. Once again this would throw a wrench into our code.

```ts
export type FieldsRateShape = {
  label: string;
  uid?: string;
  isTotal?: boolean;
  fields: {
    label: string;
    value?: string | number;
  }[];
};
```

```suggestion
        [key: string]: StandardRateShape[] | FieldsRateShape[];
```

Reply:
This is something that we'll expand when we work on the other data source calculations as there will be other areas where we might need to pull the data.